src/main/java/com/faforever/client/game/GameService.java

```diff
@@ -540,7 +541,10 @@ public Game getByUid(int uid) {
               gameLaunchMessage.getArgs().add("/team " + gameLaunchMessage.getTeam());
               gameLaunchMessage.getArgs().add("/players " + gameLaunchMessage.getExpectedPlayers());
               gameLaunchMessage.getArgs().add("/startspot " + gameLaunchMessage.getMapPosition());
-              gameLaunchMessage.getArgs().add("/gameoptions " + gameLaunchMessage.getGameOptions());
+              Map<String, String> gameOptions = gameLaunchMessage.getGameOptions();
+              gameLaunchMessage.getArgs().add("/gameoptions " + gameOptions.keySet().stream()
```

Comment:
Use entrySet over keySet. EntrySet provides an object that has both the key and associated value.

Reply:
Like this?
```java
gameLaunchMessage.getArgs().add("/gameoptions " + gameLaunchMessage.getGameOptions().entrySet().stream()
                  .map(entry -> entry.getKey() + ":" + entry.getValue())
                  .collect(Collectors.joining(" ")));